business/istio_config.go

```diff
@@ -606,33 +611,79 @@ func (in *IstioConfigService) GetIstioConfigPermissions(namespaces []string) mod
+			securityPermissions[ns] = &securityRP
+			/*
+				We can optimize this logic.
+				Instead of query all editable objects of networking.istio.io and security.istio.io we can query
+				only one per API, that will save several queries to the backend.
+
+				Synced with:
+				https://github.com/kiali/kiali-operator/blob/master/roles/default/kiali-deploy/templates/kubernetes/role.yaml#L62
+			*/
+			go func(namespace string, wg *sync.WaitGroup, networkingPermissions *models.ResourcesPermissions) {
+				defer wg.Done()
+				canCreate, canUpdate, canDelete := getPermissionsApi(in.k8s, namespace, kubernetes.NetworkingGroupVersion.Group)
+				for _, rs := range newNetworkingConfigTypes {
+					networkingRP[rs] = &models.ResourcePermissions{
+						Create: canCreate,
+						Update: canUpdate,
+						Delete: canDelete,
+					}
 				}
-				go func(namespace, resource string, permissions *models.ResourcePermissions, wg *sync.WaitGroup) {
-					defer wg.Done()
-					permissions.Create, permissions.Update, permissions.Delete = getPermissions(in.k8s, namespace, resource)
-				}(ns, rs, resourcePermissions[rs], &wg)
-			}
-			istioConfigPermissions[ns] = &resourcePermissions
+			}(ns, &wg, &networkingRP)
+
+			go func(namespace string, wg *sync.WaitGroup, securityPermissions *models.ResourcesPermissions) {
+				defer wg.Done()
+				canCreate, canUpdate, canDelete := getPermissionsApi(in.k8s, namespace, kubernetes.SecurityGroupVersion.Group)
+				for _, rs := range newSecurityConfigTypes {
+					securityRP[rs] = &models.ResourcePermissions{
+						Create: canCreate,
+						Update: canUpdate,
+						Delete: canDelete,
+					}
+				}
+			}(ns, &wg, &securityRP)
 		}
 		wg.Wait()
+
+		// Join networking and security permissions into a single result
+		for _, ns := range namespaces {
+			allRP := make(models.ResourcesPermissions, len(newNetworkingConfigTypes)+len(newSecurityConfigTypes))
+			istioConfigPermissions[ns] = &allRP
+			for resource, permissions := range *networkingPermissions[ns] {
+				(*istioConfigPermissions[ns])[resource] = permissions
+			}
+			for resource, permissions := range *securityPermissions[ns] {
+				(*istioConfigPermissions[ns])[resource] = permissions
+			}
+		}
 	}
 	return istioConfigPermissions
 }
 
 func getPermissions(k8s kubernetes.ClientInterface, namespace, objectType string) (bool, bool, bool) {
-	var canCreate, canPatch, canUpdate, canDelete bool
+	var canCreate, canPatch, canDelete bool
 	if api, ok := kubernetes.ResourceTypesToAPI[objectType]; ok {
 		resourceType := objectType
-		ssars, permErr := k8s.GetSelfSubjectAccessReview(namespace, api, resourceType, []string{"create", "patch", "update", "delete"})
+		/*
+			Kiali only uses create,patch,delete as WRITE permissions
+
+			"update" creates an extra call to the API that we know that it will always fail, introducing extra latency
+
+			Synced with:
+			https://github.com/kiali/kiali-operator/blob/master/roles/default/kiali-deploy/templates/kubernetes/role.yaml#L62
```

Comment:
just FYI - this link will eventually grow stale because its linking to "master" rather than a fixed tag. Probably should have used "v1.36.0".